crates/storage/errors/src/provider.rs

```diff
@@ -2,49 +2,50 @@ use reth_primitives::{
     Address, BlockHash, BlockHashOrNumber, BlockNumber, GotExpected, StaticFileSegment,
     TxHashOrNumber, TxNumber, B256, U256,
 };
+use derive_more::{Display, From};
+use alloc::{boxed::Box, string::String};
+use crate::db::DatabaseError;
 
 #[cfg(feature = "std")]
 use std::path::PathBuf;
 
-#[cfg(not(feature = "std"))]
-use alloc::{
-    boxed::Box,
-    string::{String, ToString},
-};
-
 /// Provider result type.
 pub type ProviderResult<Ok> = Result<Ok, ProviderError>;
 
 /// Bundled errors variants thrown by various providers.
-#[derive(Clone, Debug, thiserror_no_std::Error, PartialEq, Eq)]
+#[derive(Clone, Debug, Display, From, PartialEq, Eq)]
```

Comment:
I'd like to not use From here instead impl `From` for the few variants manually

Reply:
done

Reply:
what's the argument for this @mattsse ?

Reply:
don't like the opt-out attributes, these from impls should be opt-in